### PR TITLE
Add onSelect callback for range picker

### PR DESCRIPTION
**Pull request by Kim-Wing-Fung**
Added onSelect callback to RangePicker to match the functionality in PickerPanel.

### CODE REVIEW

**Comment by Kim-Wing-Fung** [comment]
@zombieJ @afc163

**Comment by zombieJ** [comment]
Just add param for `onCalendarChange` instead of new api pls.

**Comment by Kim-Wing-Fung** [comment]
@zombieJ Unfortunately, adding params to onCalendarChange would not allow the desired functionality and would change existing behaviour of onCalendarChange. The onSelect api is added in order to mirror the functionality of onSelect in the PickerPanel component. onCalendarChange is only triggered when a complete action is made i.e. both the start date and end date values have been entered, however the onSelect is intended to trigger when either the start date or end date values have been entered. One example of a use case is being able to disable certain end dates based on the selected start date. Since onCalendarChange only triggers when both start date and end date are entered, it is currently not possible to determine what the selected start date is before an end date has been entered or vice versa. If there are issues with introducing a new api, I am very open to alternative approaches to determine the selected start date or end date individually.

**Comment by zombieJ** [comment]
`onCalendarChange` is not `onChange`. Just check the code under your new added code:

```tsx
    if (onSelect) {
      if (!isEqual(generateConfig, getValue(mergedValue, 0), startValue)) {
        onSelect(startValue, startStr, 'start');
      }

      if (!isEqual(generateConfig, getValue(mergedValue, 1), endValue)) {
        onSelect(endValue, endStr, 'end');
      }
    }

    if (onCalendarChange) {
      onCalendarChange(values, [startStr, endStr]);
    }
```

**Comment by Kim-Wing-Fung** [comment]
@zombieJ apologies for my confusion between onCalendarChange and onChange. I will re-examine the onCalendarChange functionality. Thank you for your clarification.

**Comment by Kim-Wing-Fung** [comment]
@zombieJ as you mentioned, I've now updated the onCalendarChange parameters to incorporate additional information. This should not break existing functionality or change any APIs.

**Comment by zombieJ** [comment]
Thanks for quick response. Could you help to update info to an object like: https://github.com/react-component/picker/blob/2b5a2398918afc9b77d72947146cd31a679f7116/src/RangePicker.tsx#L68

File: src/RangePicker.tsx [hunk 1]
```diff
@@ -436,7 +440,17 @@ function InnerRangePicker<DateType>(props: RangePickerProps<DateType>) {
         : '';
 
     if (onCalendarChange) {
-      onCalendarChange(values, [startStr, endStr]);
+      const startChanged = !isEqual(generateConfig, getValue(selectedValue, 0), startValue);
```
Comment (zombieJ) [inline src/RangePicker.tsx:443]: Just use `sourceIndex` to get the updated input.

**Comment by Kim-Wing-Fung** [comment]
@zombieJ updated info as requested. Please take another look.

**Comment by zombieJ** [comment]
Merged. Could you help to update doc in antd also?

**Comment by zombieJ** [comment]
`+ rc-picker@1.12.0`

**Comment by Kim-Wing-Fung** [comment]
@zombieJ Sure, I will do a PR for antd doc soon.

**Pull request merged**